cmake_minimum_required(VERSION 2.6) breaks build on CMake ≥ 4.0

I have been testing OpenFOAM and other tools that I use on Ubuntu 26.04. I tried to compile DLBFoam as well, but it fails at the CMake step: starting from CMake 4, support for cmake_minimum_required values below 3.5 was removed. As a result, the DLBFoam unit tests build crashes with the error below:

`CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 has been removed from CMake.
  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.
  Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.`

One way to work around this is to define an environment variable, e.g. `export CMAKE_POLICY_VERSION_MINIMUM=3.5`. However, I think the minimum required version in the repository should be bumped to 3.5 at some point.

--- a/tests/validation/pyjacTests/pyjacTestMechanism/lib/CMakeLists.txt
+++ b/tests/validation/pyjacTests/pyjacTestMechanism/lib/CMakeLists.txt
@@ -1,4 +1,9 @@
-cmake_minimum_required(VERSION 2.6)
+if(CMAKE_VERSION VERSION_LESS "3.5")
+	cmake_minimum_required(VERSION 3.1)
+else()
+	# Ubuntu 26.04 ships CMake > 4, which requires project compatibility >= 3.5.
+	cmake_minimum_required(VERSION 3.5)
+endif()
 
 project(pyJac)
 
@@ -6,7 +11,7 @@ set(CMAKE_BUILD_TYPE Release)
 
 enable_language(C)
 
-#Note, this is not the most optimized compiler setup.
+# Note, this is not the most optimized compiler setup.
 set(CMAKE_C_FLAGS "-std=c99 -Ofast -fPIC")
 
 include_directories(src)

